decoder_test.go

```diff
@@ -69,20 +72,14 @@ func TestReadUint16(t *testing.T) {
 		}
 
 		number, err := d.readUint16()
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
 
 		if err == nil {
-			if number != tt.number {
-				t.Errorf("invalid number '%v' '%v'", number, tt.number)
-			}
+			assert.Equal(t, tt.number, number)
 		}
 
 		hash := d.h.Sum(nil)
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```

Comment:
Please consider using assert.Equal and put expected and actual arguments in a correct order.